### PR TITLE
Support API V2

### PR DESCRIPTION
**Pull request by Skumring**
- Add supporting HMRC API v2.0

Screenshot 2024-11-20 at 14 21 54

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -189,9 +189,14 @@ Instead of passing in the same options again and again, Valvat allows to alter i
 
 ```ruby
 Valvat.configure(
-  uk: true,
   raise_error: true,
-  http: { read_timeout: 5 }
+  http: { read_timeout: 5, proxy: { :p_addr, :p_port, :p_user, :p_pass, :p_no_proxy, :p_use_ssl } },
```
Comment (gigorok) [inline README.md]: proxy configuration should be done in a separate PR

File: README.md [hunk 2]
```diff
@@ -189,9 +189,14 @@ Instead of passing in the same options again and again, Valvat allows to alter i
...
+  uk: true,
+  hmrc: {
+    live: true, # Live mode for switching between Sandbox and Production API (TRUE by default)
```
Comment (gigorok) [inline README.md:202]: instead of adding new key - `uk` key can be used

File: README.md [hunk 3]
```diff
@@ -72,14 +72,21 @@ Valvat::Lookup.validate("DE345789003")
...
-# => true or false or nil
+Valvat::Lookup.validate(
+  "GB553557881",
+  uk: {
+    live: true,
+    client_id: '<client_id>',
+    client_secret: '<client_secret>'
+  }
+)
+# => Hash ({ client_id: '...', client_secret: '...' }) or false or nil
```
Comment (gigorok) [inline README.md]: Are these response examples valid?

Comment (Skumring) [reply README.md]: Yes, because it's about `options` for method (not global config), so for example when you pass only `client_id` & `client_secret` except/without `live`, the `live` config will be fetched from global configuration

Comment (Skumring) [reply README.md]: of course if globally uk is not set as false or nil

Comment (Skumring) [reply README.md]: sorry, my bad it's really about response, I will rollback changes
